lib/ansible/plugins/lookup/datetime.py

```diff
@@ -0,0 +1,87 @@
+# (c) 2019 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+# Make coding more python3-ish
+from __future__ import (absolute_import, division, print_function)
+
+__metaclass__ = type
+
+DOCUMENTATION = """
+    lookup: datetime
+    author: Will Thames (@willthames)
+    version_added: "2.10"
+    short_description: Creates a date time object, similar to C(ansible_date_time) fact
+    description:
+      - Creates a date time object at the time of execution. The fact is identical in
+        structure to the `ansible_date_time` fact returned from the setup module or
+        gathering facts, but can be used at the end of a playbook to calculate how
+        long something took.
+"""
+
+EXAMPLES = """
+- name: some tasks that set up the operation
+
+- name: gather the time at the start of an operation
+  set_fact:
+    start_time: "{{ lookup('datetime') }}"
+
+- name: do some really long operations
+
+- name: gather the time at the start of an operation
+  set_fact:
+    end_time: "{{ lookup('datetime') }}"
```

Comment:
The task name is incorrect (copy-paste error). Please, consider introducing a proper task name:

```yaml
- name: gather the time at the end of the operation
```

Reply:
@Teebor-Choka fixed, thank you.